src/main/java/at/hannibal2/skyhanni/features/mining/eventtracker/MiningEventDisplay.kt

```diff
@@ -0,0 +1,94 @@
+package at.hannibal2.skyhanni.features.mining.eventtracker
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.config.features.mining.MiningEventConfig
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.features.fame.ReminderUtils
+import at.hannibal2.skyhanni.utils.LocationUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.SimpleTimeMark.Companion.asTimeMark
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object MiningEventDisplay {
+    private val config get() = SkyHanniMod.feature.mining.miningEvent
+
+    private var display = mutableListOf<String>()
+
+    private val dwarvenEvents = mutableListOf<RunningEvent>()
+    private val crystalEvents = mutableListOf<RunningEvent>()
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!event.repeatSeconds(1)) return
+        updateDisplay()
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!shouldDisplay()) return
+        config.position.renderStrings(display, posLabel = "True Farming Fortune")
+    }
+
+    private fun updateDisplay() {
+        val shouldShowDwarven = when {
```

Comment:
this could be a ```when(config. showType) ```
and for the Type current, replace the true with the isInIsland
Same can be done down below